platform/ios/src/MGLMapView.mm

```diff
@@ -1948,8 +1949,9 @@ - (void)showAttribution
     
     UIViewController *viewController = self.window.rootViewController;
     if ([viewController isKindOfClass:[UINavigationController class]]) {
-        viewController = [(UINavigationController *)viewController viewControllers].firstObject;
+        viewController = ((UINavigationController *)viewController).topViewController;
     }
+    viewController = [UIViewController getTopViewControllerFrom:viewController];
```

Comment:
I may be wrong but it seems like you can remove the check for `topViewController` so maybe this can now be:

```Objective-C
UIViewController *viewController = self.window.rootViewController;
[[UIViewController getTopViewControllerFrom:viewController] presentViewController:attributionController
                                                                         animated:YES
                                                                       completion:NULL];
```

The category seems like it'll work out the `visibleViewController` (modal or otherwise).

If that works and depending on what you decide to do about https://github.com/mapbox/mapbox-gl-native/pull/8837/files#r113819069, maybe this could end up like:

```Objective-C
[[self.window.rootViewController topMostViewController] presentViewController:attributionController
                                                                     animated:YES
                                                                   completion:NULL];
```